### PR TITLE
Remove memset from call_data from gRPC server.

### PR DESCRIPTION
**Pull request by soheilhy**
This was missed due to an explicit memset(0) in init_call_elem.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.6%     +96 src/core/lib/surface/server.cc                                                           +96  +0.6%
      [NEW]    +146 request_matcher_zombify_all_pending_calls((anonymous namespace)::request_matcher*) [    +146  [NEW]
       +37%     +82 init_call_elem                                                                           +82   +37%
      +3.8%     +14 [Unmapped]                                                                               +14  +3.8%

 -+-+-+-+-+-+-+ MIXED                                                                                +-+-+-+-+-+-+-
  -0.0%     -48 [None]                                                                                  +104  +0.0%

  +0.0%     +48 TOTAL                                                                                   +200  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: src/core/lib/surface/server.cc [hunk 1]
```diff
@@ -128,38 +132,61 @@ typedef enum {
...
 
-  grpc_completion_queue* cq_new;
+  grpc_completion_queue* cq_new = nullptr;
 
-  grpc_metadata_batch* recv_initial_metadata;
-  uint32_t recv_initial_metadata_flags;
-  grpc_metadata_array initial_metadata;
+  grpc_metadata_batch* recv_initial_metadata = nullptr;
+  uint32_t recv_initial_metadata_flags = 0;
+  grpc_metadata_array initial_metadata = grpc_metadata_array();
```
Comment (markdroth) [inline src/core/lib/surface/server.cc]: This should have a comment explaining the syntax, similar to what you added in other locations in the previous PR.

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,133      Total (>)      2,019,028

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,179,800      Total (>)     11,179,738

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.6%     +96 src/core/lib/surface/server.cc                                                           +96  +0.6%
      [NEW]    +146 request_matcher_zombify_all_pending_calls((anonymous namespace)::request_matcher*) [    +146  [NEW]
       +37%     +82 init_call_elem                                                                           +82   +37%
      +3.8%     +14 [Unmapped]                                                                               +14  +3.8%

 -+-+-+-+-+-+-+ MIXED                                                                                +-+-+-+-+-+-+-
  -0.0%     -48 [None]                                                                                  +104  +0.0%

  +0.0%     +48 TOTAL                                                                                   +200  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,133      Total (>)      2,019,028

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,179,803      Total (>)     11,179,737

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by soheilhy** [comment]
TSAN failure is unrelated #16896 which I sent a separate fix for in #17202

**Pull request merged**